src/answers.sql

```diff
@@ -1,15 +1,119 @@
 -- Your answers here:
 -- 1
+SELECT 
+    c.name AS country_name, 
+    COUNT(s.id) AS total_states
+FROM 
+    states s
+JOIN 
+    countries c
+ON 
+    s.country_id = c.id
+GROUP BY 
+    c.name;
 
 -- 2
+SELECT COUNT(*) AS employees_wothout_bosses
+FROM employees
+WHERE supervisor_id IS NULL;
 
 -- 3
+    SELECT 
+    c.name AS country_name, 
+    o.address AS address, 
+    COUNT(e.id) AS numer_employees
+FROM 
+    offices o
+JOIN 
+    employees e ON o.id = e.office_id
+JOIN 
+    countries c ON o.country_id = c.id	
+GROUP BY 
+    c.name, o.address
+ORDER BY 
+    numer_employees DESC, c.name ASC
+LIMIT 5;
 
 -- 4
+SELECT 
+    s.id AS supervisor_id,
+    COUNT(e.id) AS number_of_employe
+FROM 
+    employees e
+JOIN 
+    employees s ON e.supervisor_id = s.id
+GROUP BY 
+    s.id
+ORDER BY 
+    number_of_employe DESC
+LIMIT 3;
+
 
 -- 5
+SELECT 
+    s.name AS state_name, 
+    COUNT(o.id) AS colorado_offices
+FROM 
+    offices o
+JOIN 
+    states s ON o.state_id = s.id
+GROUP BY 
+    s.name
+HAVING 
+    s.name = 'Colorado';
```

Comment:
Here we might use `WHERE` clause.
https://www.sql-easy.com/learn/how-to-use-having-clause-in-sql/